File: src/strategies/ce-sma.ts
import { Strategy } from "@/classes/strategy";
import { ICandle } from "@/utils/interfaces";
const fastRSI = 30
export class RSI_ONLY extends Strategy {
    name: string = "RSI ONLY";
    desc: string = `Enter: macd > 0 && sma20 >  sma50, Exit: oposite`;

    buyCond(row: ICandle): boolean {
        return (row.rsi < fastRSI)
    }

    sellCond(row: ICandle): boolean {
        return row.rsi > 100 - fastRSI
    }
}
export class ANY extends Strategy {
    name: string = "ANY";
    desc: string = `Enter: macd > 0 && sma20 >  sma50, Exit: oposite`;

    buyCond(row: ICandle): boolean {
        return true
    }

    sellCond(row: ICandle): boolean {
        return true
    }
}
class BB_SMA extends Strategy {
    name: string = "BB_SMA";
    desc: string = `Enter: macd > 0 && sma20 >  sma50, Exit: oposite`;

    buyCond(row: ICandle): boolean {
        const smaCond = row.sma_20 >= row.sma_50 && row.macd >= 0;
        return row.c > row.o || smaCond;
    }

    sellCond(row: ICandle): boolean {
        const smaCond = row.sma_20 <= row.sma_50 && row.macd <= 0;
        return row.c < row.o || smaCond;
    }
}
class ThreeSum extends Strategy {
    name: string = "ThreeSum";
    desc: string = `Enter: macd > 0 && sma20 >  sma50, Exit: oposite`;

    buyCond(row: ICandle, df?: ICandle[], i?: number): boolean {
        const smaCond = row.buy_signal == 1 && row.sma_20 > row.sma_50
                let bool = true;
        return smaCond && bool;
    }

    sellCond(row: ICandle, entry?: number, df?: ICandle[], i?: number): boolean {
        const smaCond = row.sell_signal == 1 && row.sma_20 < row.sma_50
               let bool = true;
        return smaCond && bool;
    }
}

export const strategies = [new ANY()]// [ new BB_SMA(), new ThreeSum()];

File: strategies/ce_sma.py

# Define specific strategy classes
from typing import List
from classes.index import Strategy

fast_rsi = 50

class RSI_ONLY(Strategy):
    def __init__(self):
        self.name = "RSI ONLY"
        self.desc = "Enter: RSI < fastRSI, Exit: RSI > 100 - fastRSI"

    def buy_cond(self, row: dict) -> bool:
        return row['rsi'] < fast_rsi

    def sell_cond(self, row: dict) -> bool:
        return row['rsi'] > 100 - fast_rsi

class ANY(Strategy):
    def __init__(self):
        self.name = "ANY"
        self.desc = "Enter: True, Exit: True"

    def buy_cond(self, row: dict) -> bool:
        return True

    def sell_cond(self, row: dict) -> bool:
        return True

# Add more strategy classes as needed

# List of strategies
strategies: List[Strategy] = [
    RSI_ONLY(),
    ANY()
    # Add more strategy instances here if needed
]